packages/bpk-component-chip-group/src/BpkChipGroup.tsx

```diff
@@ -0,0 +1,213 @@
+ */
+import type { ReactElement, ReactNode} from 'react';
+import { useRef, useState } from 'react';
+
+import BpkBreakpoint, { BREAKPOINTS } from '../../bpk-component-breakpoint';
+import BpkSelectableChip, { BpkDismissibleChip, BpkIconChip, BpkDropdownChip, CHIP_TYPES } from '../../bpk-component-chip';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import FilterIconSm from '../../bpk-component-icon/sm/filter';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import BpkMobileScrollContainer from '../../bpk-component-mobile-scroll-container';
+import { cssModules } from '../../bpk-react-utils';
+
+import Nudger from './Nudger';
+
+import STYLES from './BpkChipGroup.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export const CHIP_GROUP_TYPES = {
+  rail: 'rail',
+  wrap: 'wrap',
+};
+
+export const CHIP_COMPONENT = {
+  selectable: 'selectable',
+  dismissible: 'dismissible',
+  dropdown: 'dropdown',
+  icon: 'icon',
+};
+
+const CHIP_COMPONENT_MAP = {
+  [CHIP_COMPONENT.selectable]: BpkSelectableChip,
+  [CHIP_COMPONENT.dismissible]: BpkDismissibleChip,
+  [CHIP_COMPONENT.dropdown]: BpkDropdownChip,
+  [CHIP_COMPONENT.icon]: BpkIconChip,
+}
+
+export type ChipGroupType = (typeof CHIP_GROUP_TYPES)[keyof typeof CHIP_GROUP_TYPES];
+export type ChipStyleType = (typeof CHIP_TYPES)[keyof typeof CHIP_TYPES];
+export type ChipComponentType = (typeof CHIP_COMPONENT)[keyof typeof CHIP_COMPONENT];
+
+export type SingleSelectChipItem = {
+  text: string;
+  accessibilityLabel?: string;
+  leadingAccessoryView?: ReactNode;
+  [rest: string]: any; // Inexact rest. See decisions/inexact-rest.md
+};
+
+export type ChipItem = {
+  component?: ChipComponentType;
+  onClick?: (selected: boolean, index: number) => void;
+  selected?: boolean;
+  hidden?: boolean;
+} & SingleSelectChipItem;
+
+export type CommonProps = {
+  ariaLabel?: string;
+  type?: ChipGroupType;
+  chipStyle?: ChipStyleType;
+  // only required when type = rail
+  leadingNudgerLabel?: string;
+  trailingNudgerLabel?: string;
+};
+
+export type ChipGroupProps = {
+  chips: ChipItem[];
+  stickyChip?: ChipItem;
+  ariaMultiselectable?: boolean;
+} & CommonProps;
+
+
+const BpkChipGroup = ({
+  ariaLabel,
+  ariaMultiselectable = true,
+  chipStyle = CHIP_TYPES.default,
+  chips,
+  leadingNudgerLabel = '',
+  stickyChip,
+  trailingNudgerLabel = '',
+  type = CHIP_GROUP_TYPES.rail,
+}: ChipGroupProps) => {
+  const scrollContainerRef = useRef<HTMLElement | null>(null);
+
+  const containerClassnames = getClassName('bpk-chip-group-container')
+
+  const chipGroupClassNames = getClassName(
+    'bpk-chip-group',
+    `bpk-chip-group--${type}`,
+  );
+
+  const stickyChipContainerClassnames = getClassName(
+    'bpk-sticky-chip-container',
+    `bpk-sticky-chip-container--${chipStyle}`,
+  );
+
+  const scrollContainerIndicator = getClassName('bpk-chip-group-scroller-indicator');
+
+  const renderChipItem = ({
```

Comment:
Is there a reason that this needs to be inline within `BpkChipGroup`. Looks like we can add `chipStyle` to the props and extract it from the definition. 

The reason that we might want to do this is that as written a new function `renderChipItem` will be created on each render of `BpkChipGroup` and we can just avoid it. 

This could also allow us to turn `renderChipItem` into a component which could make the code more readable? Certainly less surprising seeing a component than a function _that happens to return JSX_, if you see what I mean.